rotkehlchen/exchanges/kucoin.py

```diff
@@ -403,6 +402,12 @@ def _api_query_paginated(
             except KeyError as e:
                 msg = f'Kucoin {case} JSON response is missing key: {str(e)}'
                 log.error(msg, response_dict)
+                if case == KucoinCase.OLD_TRADES and '400100' == response_dict.get('code', ''):
+                    if current_query_ts + time_step >= end_ts:
```

Comment:
I've been checking and this logic is a bit complicated. If I move this logic to the start of the loop It will conflict with the pagination that we do in the same loop